http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -115,33 +116,39 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         }
       }
 
-      // Create WSM managed identity if shared app
+      wsmResourceApi <- buildWsmResourceApiClient
+
+      // Create or fetch WSM managed identity if shared app
       wsmManagedIdentityOpt <- app.samResourceId.resourceType match {
-        case SamResourceType.SharedApp =>
-          childSpan("createWsmIdentityResource").use { implicit ev =>
-            createWsmIdentityResource(app, namespacePrefix, params.workspaceId).map(_.some)
+        // for apps that support cloning, if a managed identity has already been created in the workspace
+        // use that otherwise create a new managed identity
+        case SamResourceType.SharedApp if doesAppTypeSupportCloning(app.appType) =>
```

Comment:
Just curious, do we need to check `doesAppTypeSupportCloning`? Could we just always try to retrieve an existing identity for the app type, and use it if it exists? Just thinking that might be more flexible, if we need to support other cloneable apps in the future.

Reply:
I had thought about this and added `doesAppTypeSupportCloning` check to avoid unnecessary calls to WSM for other apps that don't support cloning. But I am happy to remove that change if you think it is overkill.

Reply:
I think I would value less complexity/maintenance over optimization in this case. These are all async operations, and we are already making a ton of requests to WSM.

Reply:
Sounds good. I will remove that check.